### PR TITLE
Improve Documentation!

### PR DESCRIPTION
**Pull request by dcookman**
A push to add a file explaining what the heck is going on with OXO was first done in PR #24.
This new PR builds on this, adding explanations for almost all of the other classes within the OXO framework.
Major additions include explanations on `TestStatistic`s, `Optimiser`s, and all of the "manager" classes we have floating around.

It's not perfect, but hopefully this can be a major boost to new folks trying to work out what on Earth is going on with the codebase, as well as old hands trying to remember what the difference between a `FitComponent` and a `FitParameter` was again.

### CODE REVIEW

**Review by willp240 — APPROVED** [review]
I've not read it yet (of course I plan to!) but I'm inclined to just get this merged in. We can always modify later if we want to but more documentation is always better than less!

File: HOWTO.md [hunk 1]
```diff
@@ -214,17 +272,27 @@ dimensions it is defined on, with a possible position of the step for each
...
 `Integrate(mins, maxs)` method. Because a `PDF` has a well-defined integral,
 it can also be also randomly sampled from: `Sample()`. See the section on 
 [random number generation](#4-random-number-generation-rand) for how derived
 classes implement this.
 
-##### 3.2.1.2.1. Gaussian
+##### 3.2.1.3.1. Gaussian
 For all of that effort, we can finally make some actual PDFs, in particular a
 multi-dimensional Gaussian (given how useful they are!). Unsurprisingly, 
 `Gaussian` derives from the `PDF` class.
```
Comment (willp240) [inline HOWTO.md:298]: on line 305 I think handles -> handled (It wasn't changed in this PR so can't comment on the exact right line)

File: HOWTO.md [hunk 2]
```diff
@@ -318,7 +438,8 @@ pretend sample of data, particularly useful for sensitivity studies? Well, the
 `DataSetGenerator` class has you covered. Add `DataSet` objects along with their
 expected rates to the generator object. The `sequential` flag for each data set
 determines whether generation will use the events in each `DataSet` sequentially.
-If set to false, events are selected randomly. There is also a set of `bootstrap` flags, which if set to true for a given `DataSet` will draw randomly from the
+If set to false, events are selected randomly. There is also a set of `bootstrap` flags, 
+which if set to true for a given `DataSet` will draw randomly from the
 dataset with replacement: useful in a pinch if you need to have a fake dataset
 of size larger than your MC!
```
Comment (willp240) [inline HOWTO.md:445]: Line 454 space between 'B-types.' and 'Both'

File: HOWTO.md [hunk 3]
```diff
@@ -342,6 +463,7 @@ back the events not used for previous event generation from the ith `DataSet`.
 Finally, one can also provide cuts on what can be generated by `SetCuts()` or
 `AddCut()`; the event rates the user gives then correspond to the event rates
 foreach dataset *before* cuts. More on cuts in OXO [here](#61-cut).
+
 ## 4.2. BinnedEDGenerator
 The `BinnedEDGenerator` works very much like `DataSetGenerator`, except the input
 and output objects are no longer `DataSets` which contain sets of `Event` objects,
```
Comment (willp240) [inline HOWTO.md:469]: Line 632 capital This

File: HOWTO.md [hunk 4]
```diff
@@ -516,10 +638,22 @@ shift parameter.
...
+of an event are classic examples. Instead of doing more convolutions, this class 
+just modifies the post-smear value linearly relative to the pre-smear value: 
+`post_smear_new = pre_smear + fCorrection*(post_smear_old - pre_smear)`. 
+
+This class has one parameter, `correction`, which defines the smearing scale. 
+If `correction = 1`, nothing changes. If `correction = 0`, all smearing is removed 
+and the post-smear values will equal that of pre-smear.
 
 ## 7.2. Systematic
 Unlike `EventSystematic`, the `Systematic` class is for manipulating a binned
```
Comment (willp240) [inline HOWTO.md:659]: Line 678 space between 'upon.' and 'These'

File: HOWTO.md [hunk 5]
```diff
@@ -549,16 +683,605 @@ The `Scale` systematic works like `EventScale`, except now of course the impact
...
+To solve this problem, there is a second approach to handle parameters. It is more 
+complicated, but in the end we will see that it grants `FitComponent` status to 
+collections of `BinnedED` objects, for example.
+
+To start with, the `FitParameter` abstract base class defines the concept of a 
+single parameter, which merely has the `Get()` and `Set()` methods.
+
+NOTE: `FitComponent` and `FitParameter` are totally different things! Both are abstract 
+base classes, but `FitParameter` describes a single parameter, whereas `FitComponent` is 
+any object that has fittable components. Sorry.
```
Comment (willp240) [inline HOWTO.md:783]: 😄

File: HOWTO.md [hunk 6]
```diff
@@ -549,16 +683,605 @@ The `Scale` systematic works like `EventScale`, except now of course the impact
...
+normalisation is a `FitParameter`, and there is a separate pos-systematic/shrinking 
+normalisation that can be acquired through `GetNormalisations()`.
+
+Finally, you might have a PDF which you want to allow a systematic object to control the 
+normalisation directly, and have the pre-systematic normalisation variable within 
+the `BinnedEDManager` turned off. This is the `FALSE` mode.
+
+
+# 10. Other Managers
+In the previous section, a bunch of "Manager" vlasses were explained. Well, it turns out
```
Comment (willp240) [inline HOWTO.md:877]: vlasses -> classes

File: HOWTO.md [hunk 7]
```diff
@@ -549,16 +683,605 @@ The `Scale` systematic works like `EventScale`, except now of course the impact
...
+`AddDist(pdf, group_name)` methods. You can then associate a systematic with a group when 
+adding it to be tracked by the manager: `Add(sys, group_name)`. If a systematic is added 
+without a group, then it is "global", and is applied to all PDFs.
+
+## 10.4. BinnedEDShrinker
+The `BinnedEDShrinker` isn't really a "manager", but it's worth talking about at this point 
+anyways. When applying systematics to PDFs, problems can rapidly arise near the edges of 
+the PDFs. In order to avoid this issue, it is generally wise to put "buffer" bins around the 
+edges of the PDFs. The buffer bins continue to store PDF information as usual, so that 
+systematics can stil be applied in the region, but are then sacrificed during the final
```
Comment (willp240) [inline HOWTO.md:926]: stil -> still

**Comment by dcookman** [comment]
@willp240 just fixed those typos!

**Comment by willp240** [comment]
Awesome thanks Daniel! Let's merge in the call tomorrow!

**Pull request merged**